src/email-only-signup.js

```diff
@@ -1,132 +1,284 @@
 		}
+	}
+
+	// toggle the email validation errors
+	function toggleComponentValidationErrors () {
+		toggleClass(FORM_ERROR_CLASS, o.emailField);
+		toggleClass(VISUALLY_HIDDEN_CLASS, o.invalidEmailMessage);
+	}
+
+	// collapse component, aka. show discreet state
+	function showComponentCollapsed (showCollapsed) {
 
-		function encodeComponent(string) {
-			return encodeURIComponent(string.trim()).replace('%20', '+');
+		o.content.setAttribute('aria-hidden', showCollapsed);
+		toggleClass(VISUALLY_HIDDEN_CLASS, o.content, showCollapsed);
+		_forEach(o.contentFocusables, el => {
+			toggleTabIndex(el, !showCollapsed);
+		});
+
+		o.discreetContent.setAttribute('aria-hidden', !showCollapsed);
+		toggleClass(VISUALLY_HIDDEN_CLASS, o.discreetContent, !showCollapsed);
+		_forEach(o.discreetContentFocusables, el => {
+			toggleTabIndex(el, showCollapsed);
+		});
+
+		updateComponentAriaControls();
+	}
+
+	// toggle the inactive styles on dropdown component to allow mock 'placeholder'
+	function toggleComponentSelectInactive (event) {
+		let isPlaceholderSelected = (event.target.options[event.target.selectedIndex].getAttribute('placeholder') !== null);
+
+		if (event.type === 'focus') {
+			removeClass(SELECT_INACTIVE_CLASS, o.topicSelect);
 		}
 
-		function optionsFromData(el) {
-			const options = {};
-			Object.keys(defaultOptions).forEach(key => {
-				// convert optionKeyLikeThis to data-o-email-only-signup-option-key-like-this
-				const attr = 'data-o-email-only-signup-' + kebabCase(key);
-				if(el.hasAttribute(attr)) {
-					options[key] = el.getAttribute(attr);
-				}
-			});
-
-			return options;
+		if (event.type === 'blur' && isPlaceholderSelected) {
+			addClass(SELECT_INACTIVE_CLASS, o.topicSelect);
 		}
+	}
 
-		function serializeFormInputs(form) {
-			const inputs = form.elements;
-			let str = [];
+	/*
+	 * display a message in the component
+	 * @param {HTML String} message - content to inject
+	 */
+	function displayComponentMessage (message) {
+		o.displaySection.innerHTML = message;
+	}
 
-			for (let i=0; i<inputs.length; i++) {
-				let field = form.elements[i];
-				if (field.name && field.type !== 'submit' && field.type !== 'button') {
-					str.push(`${encodeComponent(field.name)}=${encodeComponent(field.value)}`);
-				}
+	//update 'aria-expanded' attr to match an 'aria-controls' element target state
+	function updateComponentAriaControls () {
+		_forEach(o.ariaControls, el => {
+			const target = o.self.querySelector('#' + el.getAttribute('aria-controls'));
+			if (target) {
+				const targetIsHidden = (target && target.getAttribute('aria-hidden') === 'true');
+				el.setAttribute('aria-expanded', !targetIsHidden);
 			}
+		});
+	}
+};
 
-			return str.join("&");
+/*
+ * Extract options from element data attributes
+ * @param {Element} el
+ * @param {Object} options - Containing keys to match attrs
+ * @returns {Object}
+ */
+function optionsFromData (el, opts) {
+	const options = {};
+	_forEach(Object.keys(opts), key => {
```

Comment:
Any particular reason you've gone for lodash here? I would assume any product supporting a non-ES5 browser would already be using the polyfill service?